manticore/ethereum.py

```diff
@@ -697,14 +735,15 @@ def _compile(source_code, contract_name):
 
             if contract['bin'] == '':
                 raise Exception('Solidity failed to compile your contract.')
-                
-            bytecode = contract['bin'].decode('hex')
+
+            bytecode = ManticoreEVM._link(contract['bin'], libraries).decode('hex')
             srcmap = contract['srcmap'].split(';')
             srcmap_runtime = contract['srcmap-runtime'].split(';')
             hashes = contract['hashes']
             abi = json.loads(contract['abi'])
-            runtime = contract['bin-runtime'].decode('hex')
+            runtime = ManticoreEVM._link(contract['bin-runtime'], libraries).decode('hex')
```

Comment:
`str.decode` doesn't exist in Py3 anymore. Use `import codecs` and `codecs.decode`